prowler/providers/aws/services/iam/lib/policy.py

```diff
@@ -68,3 +73,96 @@ def is_policy_public(policy: dict) -> bool:
                     ) and "Condition" not in statement:
                         return True
     return False
+
+
+def check_full_service_access(service: str, policy: dict) -> bool:
+    """
+    check_full_service_access checks if the policy allows full access to a service.
+    Args:
+        service (str): The service to check.
+        policy (dict): The policy to check.
+    Returns:
+        bool: True if the policy allows full access to the service, False otherwise.
+    """
+
+    full_access = False
+
+    if policy:
+        policy_statements = policy.get("Statement", [])
+
+        if not isinstance(policy_statements, list):
+            policy_statements = [policy["Statement"]]
+
+        for statement in policy_statements:
+            if statement.get("Effect", "") == "Allow":
+                resources = statement.get("Resource", [])
+
+                if not isinstance(resources, list):
+                    resources = [statement.get("Resource", [])]
+
+                if "*" in resources:
+                    if "Action" in statement:
+                        actions = statement.get("Action", [])
+
+                        if not isinstance(actions, list):
+                            actions = [actions]
+
+                        if f"{service}:*" in actions:
+                            full_access = True
+                            break
+
+                    elif "NotAction" in statement:
+                        not_actions = statement.get("NotAction", [])
+
+                        if not isinstance(not_actions, list):
+                            not_actions = [not_actions]
+
+                        if f"{service}:*" not in not_actions:
+                            full_access = True
+                            break
+
+    return full_access
+
+
+def is_condition_restricting_from_private_ip(condition_statement: dict) -> bool:
+    """Check if the policy condition is coming from a private IP address.
+
+    Keyword arguments:
+    condition_statement -- The policy condition to check. For example:
+        {
+            "IpAddress": {
+                "aws:SourceIp": "X.X.X.X"
+            }
+        }
+    """
+
+    is_from_private_ip = False
+
+    if condition_statement.get("IpAddress", {}):
```

Comment:
I will move this line into the function caller, makes little sense to call the function if the condition is not `IpAddress`.